File: ping_noc_.py
import socket
import time
import argparse

def main():
    # Configuración del cliente
    host = input("Ingrese la dirección IP del servidor: ")
    port = int(input("Ingrese el puerto del servidor: "))

    # Crear un socket UDP/IP
    client_socket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)

    print(f"Cliente UDP Ping conectado al servidor {host}:{port}")

    # Crear un objeto ArgumentParser
    parser = argparse.ArgumentParser(description="Enviar paquetes ICMP a un servidor.")

    # Añadir el argumento para el número de paquetes
    parser.add_argument("--num-paquetes", type=int, default=float('inf'), help="Número de paquetes a enviar.")

    # Parsear los argumentos
    args = parser.parse_args()

    # Número de paquete inicial
    packet_number = 0

    while packet_number < args.num_paquetes:
        # Incrementar el número de paquete
        packet_number += 1

        # Obtener la marca de tiempo antes de enviar el mensaje
        start_time = time.time()

        # Construir el mensaje a enviar al servidor (simula el número de paquete)
        message = str(packet_number)

        # Enviar datos al servidor
        client_socket.sendto(message.encode(), (host, port))

        # Recibir respuesta del servidor
        data, server_address = client_socket.recvfrom(1024)
        print(f"Respuesta del servidor: {data.decode()}")

        # Calcular la diferencia de tiempo después de recibir la respuesta
        end_time = time.time()
        elapsed_time = end_time - start_time

        # Imprimir el tiempo transcurrido
        print(f"Tiempo transcurrido: {elapsed_time} segundos")

        # Esperar un segundo antes de enviar el siguiente paquete (simula el ping)
        time.sleep(1)

    # Cerrar el socket del cliente
    client_socket.close()

if __name__ == "__main__":
    main()

File: ping_noc.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>
#include <arpa/inet.h>

#define MAX_BUF_SIZE 1024

int main() {
    int client_socket;
    struct sockaddr_in server_addr;
    char message[MAX_BUF_SIZE];
    int port;
    char ip[16];

    // Solicitar al usuario ingresar la dirección IP del servidor
    printf("Ingrese la dirección IP del servidor: ");
    scanf("%s", ip);

    // Solicitar al usuario ingresar el puerto del servidor
    printf("Ingrese el puerto del servidor: ");
    scanf("%d", &port);

    // Crear el socket UDP
    client_socket = socket(AF_INET, SOCK_DGRAM, 0);
    if (client_socket < 0) {
        perror("Error al crear el socket");
        exit(EXIT_FAILURE);
    }

    // Configurar la dirección del servidor
    memset(&server_addr, 0, sizeof(server_addr));
    server_addr.sin_family = AF_INET;
    server_addr.sin_port = htons(port);
    server_addr.sin_addr.s_addr = inet_addr(ip);

     int icmp_seq = 0;
    while (1) {
        // Incrementar el número de secuencia ICMP
        
        icmp_seq++;

        // Construir el mensaje ICMP simulado
        snprintf(message, MAX_BUF_SIZE, "Paquete ICMP_SEQ=%d", icmp_seq);

        // Enviar el mensaje al servidor
        if (sendto(client_socket, message, strlen(message), 0, (struct sockaddr*)&server_addr, sizeof(server_addr)) < 0) {
            perror("Error al enviar el mensaje al servidor");
            exit(EXIT_FAILURE);
        }

        // Esperar la respuesta del servidor
        memset(message, 0, sizeof(message));
        recvfrom(client_socket, message, MAX_BUF_SIZE, 0, NULL, NULL);

        // Imprimir la respuesta del servidor
        printf("Respuesta del servidor: %s\n", message);

        // Esperar un tiempo antes de enviar el próximo paquete (opcional)
        sleep(1);
        
    }

    close(client_socket);
    return 0;
}
